caledonia/src/prehashed.rs

```diff
@@ -0,0 +1,331 @@
+//! Rust implementation of ALBA's prehashed scheme using Blake2b as hash
+//! function, working for big sets.
+
+extern crate core;
+use crate::utils;
+
+use std::f64::consts::E;
+
+const DATA_LENGTH: usize = 32;
+const DIGEST_SIZE: usize = 32;
+
+type Data = [u8; DATA_LENGTH];
+type Hash = [u8; DIGEST_SIZE];
+
+/// Setup input parameters
+#[derive(Debug, Clone)]
+pub struct Params {
```

Comment:
`Params` is common with `bounded.rs` and we could maybe extract the common functionality in a trait?